pkg/reconciler/pipelinerun/resources/apply_test.go

```diff
@@ -1450,6 +1450,125 @@ func TestApplyParameters(t *testing.T) {
 			}},
 		},
 	},
+		{
+			name: "tasks with the same parameter name but referencing different values",
+			original: v1beta1.PipelineSpec{
+				Params: []v1beta1.ParamSpec{
+					{
+						Name: "start",
+						Default: &v1beta1.ParamValue{
+							Type:      v1beta1.ParamTypeString,
+							StringVal: "a",
+						},
+						Type: v1beta1.ParamTypeString,
+					},
+				},
+				Tasks: []v1beta1.PipelineTask{
+					{
+						Name: "print-msg",
+						TaskSpec: &v1beta1.EmbeddedTask{
+							TaskSpec: v1beta1.TaskSpec{
+								Params: []v1beta1.ParamSpec{
+									{
+										Name: "start",
+										Type: v1beta1.ParamTypeString,
+									},
+								},
+							},
+						},
+						Params: []v1beta1.Param{
+							{
+								Name: "start",
+								Value: v1beta1.ParamValue{
+									Type:      v1beta1.ParamTypeString,
+									StringVal: "$(tasks.start.results.Output)",
```

Comment:
It might make sense to include a "start" task in the test case-- otherwise this param value is sort of confusing?

Reply:
thanks @lbernick , I added a `previous-task-with-result` task, and changed the param `$(tasks.start.results.Output)` to `$(tasks.previous-task-with-result.results.Output)`, maybe it looks clearer